apps/plugins/pkg/app/install/registrar.go

```diff
@@ -0,0 +1,163 @@
+package install
+
+import (
+	"context"
+	"sync"
+
+	"github.com/grafana/grafana-app-sdk/resource"
+	errorsK8s "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	pluginsv0alpha1 "github.com/grafana/grafana/apps/plugins/pkg/apis/plugins/v0alpha1"
+)
+
+const (
+	PluginInstallSourceAnnotation = "plugins.grafana.app/install-source"
+)
+
+// Class represents the plugin class type in an unversioned internal format.
+// This intentionally duplicates the versioned API type (PluginInstallSpecClass) to decouple
+// internal code from API version changes, making it easier to support multiple API versions.
+type Class = string
+
+const (
+	ClassCore     Class = "core"
+	ClassExternal Class = "external"
+	ClassCDN      Class = "cdn"
+)
+
+type Source = string
+
+const (
+	SourceUnknown     Source = "unknown"
+	SourcePluginStore Source = "plugin-store"
+)
+
+type PluginInstall struct {
+	ID      string
+	Version string
+	URL     string
+	Class   Class
+	Source  Source
+}
+
+func (p *PluginInstall) ToPluginInstallV0Alpha1(namespace string) *pluginsv0alpha1.PluginInstall {
+	var url *string = nil
+	if p.URL != "" {
+		url = &p.URL
+	}
+	return &pluginsv0alpha1.PluginInstall{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      p.ID,
+			Annotations: map[string]string{
+				PluginInstallSourceAnnotation: p.Source,
```

Comment:
The source at the moment is alwasy "plugin-store", what are other possible values here?

Reply:
sorry, i missed this one. it will eventually include external sync mechanisms like the stack state service